cmd/medius/images/push.go

```diff
@@ -107,45 +108,33 @@ func NewPublishImagesCommand(options *common.Options) *cobra.Command {
 }
 
 func (b *buildAndPublish) Do(entry *common.Entry, timestamp time.Time) ([]string, error) {
-	metadata := entry.Artifact.Metadata()
-	artifactInfo, err := entry.Artifact.Inspect()
+	metadata := entry.Artifacts[0].Metadata()
+	artifactInfo, err := entry.Artifacts[0].Inspect()
 	if err != nil {
 		return nil, fmt.Errorf("error introspecting artifact %q: %v", metadata.Describe(), err)
 	}
-	b.Log.Infof("Remote artifact checksum: %q", artifactInfo.SHA256Sum)
 
-	imageSha, err := b.getImageSha(metadata.Describe())
+	images, artifacts, err := b.buildImages(entry)
 	if err != nil {
 		return nil, err
 	}
-	if imageSha == artifactInfo.SHA256Sum && !b.Options.PublishImagesOptions.ForceBuild {
-		b.Log.Info("Nothing to do.")
-		return nil, nil
-	}
-	if errors.Is(b.Ctx.Err(), context.Canceled) {
-		return nil, b.Ctx.Err()
-	}
+	defer cleanupArtifacts(artifacts)
 
-	b.Log.Infof("Rebuild needed, downloading %q ...", artifactInfo.DownloadURL)
-	file, err := b.getArtifact(artifactInfo)
+	containerDiskIndex, err := build.ContainerDiskIndex(images)
```

Comment:
Move this to L131, we don't need to do it if we're pushing a single image.

Reply:
Done.